__tests__/app.test.js

```diff
@@ -160,14 +160,40 @@ describe("/api/articles", () => {
         expect(articles).toBeSortedBy("author", { descending: true });
       });
   });
-  test("GET 400: should recieve an error when sort_by is given a column that doesn't exist", () => {
+  test("GET 400: responds with an error when sort_by is given a column that doesn't exist", () => {
     return request(app)
       .get("/api/articles?sort_by=non_existent")
       .expect(400)
       .then(({ body }) => {
         expect(body.msg).toBe("Bad Request: Invalid sort_by!");
       });
   });
+  test("GET 400: responds with an error when an order is invalid", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at&order=invalid_order")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request: Invalid order!");
+      });
+  });
+  test("GET 200: responds with articles which have been filtered based on topic query", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        articles.forEach((article) => {
+          expect(article.topic).toBe("cats");
+        });
+      });
+  });
+  test("GET 404: responds with an error when no article matches the topic query", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
```

Comment:
the topic exists, just that there are no articles.  feedback how you want to the user! but theres an argument for this being a 200!